lib/elixir/lib/registry.ex

```diff
@@ -1136,6 +1142,55 @@ defmodule Registry do
     end
   end
 
+  @doc """
+  Allows selecting key, pid and values registered using full match specs.
+
+  Note that for large registries with many partitions this will be costly as it builds the list by
+  concatenating all the partitions.
+
+  ## Examples
+
+  This example shows how to get everything from the registry.
+
+      iex> Registry.start_link(keys: :unique, name: Registry.SelectAllTest)
+      iex> {:ok, _} = Registry.register(Registry.SelectAllTest, "hello", :value)
+      iex> {:ok, _} = Registry.register(Registry.SelectAllTest, "world", :value)
+      iex> Registry.select(Registry.SelectAllTest, [{{:"$1", :"$2", :"$3"}, [], [{{:"$1", :"$2", :"$3"}}]}])
+      [{"world", self(), :value}, {"hello", self(), :value}]
+
+  Get all keys in the registry.
+
+      iex> Registry.start_link(keys: :unique, name: Registry.SelectAllTest)
+      iex> {:ok, _} = Registry.register(Registry.SelectAllTest, "hello", :value)
+      iex> {:ok, _} = Registry.register(Registry.SelectAllTest, "world", :value)
+      iex> Registry.select(Registry.SelectAllTest, [{{:"$1", :_, :_}, [], [:"$1"]}]) |> Enum.sort()
+      ["hello", "world"]
+
+  """
+  @doc since: "1.x.x"
+  @spec select(registry, spec) :: [term]
+  def select(registry, spec) do
+    spec = to_internal_spec(spec)
```

Comment:
Since this can be a list of specs, we need to do something like:

```
for {match, guards, return} <- specs do
  case match  do
    {_, _, _} -> match
    _ -> raise "..."
  end
end
```

We should probably validate that return does not contain "$_" too, for the reasons you mentioned.

Reply:
I'm afraid this validacion can become very complicated because of the match spec format itself and then there are guards as well. I don’t think it will be practical to do this.

Reply:
You mean the validation as a whole or the validation of the `return` part?

Reply:
Of the return part - in theory it can contain arbitrarily nested lists, tuples maps, call various ets-safe functions and probably a lot more. I don't believe we'd be able to get the validation right without some major effort.